fissa/core.py

```diff
@@ -274,16 +282,20 @@ def separation_prep(self, redo=False):
             been run. Default is `False`.
 
         """
-        # define filename where data will be present
-        fname = self.folder + '/preparation.npy'
+        # if data_format is defined, store or load the data
+        if self.data_format is None:
+            redo = True
+        else:
+            # define filename where data will be present
+            fname = self.folder + '/preparation.'+self.data_format
 
-        # try to load data from filename
-        if not redo:
-            try:
-                nCell, raw, roi_polys = np.load(fname)
-                print('Reloading previously prepared data...')
-            except BaseException:
-                redo = True
+            # try to load data from filename
+            if not redo:
+                try:
+                    nCell, raw, roi_polys = np.load(fname)
+                    print('Reloading previously prepared data...')
+                except BaseException:
+                    redo = True
```

Comment:
Remove superfluous nesting.
```suggestion
        # try to load data from filename
        if not redo:
            try:
                nCell, raw, roi_polys = np.load(fname)
                print('Reloading previously prepared data...')
            except BaseException:
                redo = True
```